bindings/python/pymongoarrow/pandas_types.py

```diff
@@ -147,7 +147,10 @@ def isna(self):
         )
 
     def __eq__(self, other):
-        return self.data == other
+        try:
+            return self.data == other
+        except Exception:
+            return self.data == np.array(other, dtype=object)
```

Comment:
Exception seems very broad here. Is there something more specific we can use?

Reply:
Fixed to be more precise